kartothek/io/dask/dataframe.py

```diff
@@ -301,3 +304,66 @@ def update_dataset_from_ddf(
         metadata=metadata,
         metadata_merger=metadata_merger,
     )
+
+
+def collect_dataset_statistics(
+    store, dataset_uuid, table_name, predicates=None, frac=0.3, factory=None,
+):
+    """
+    Retrieve some statistics, maybe even plot them, generate a report...?
+
+    Parameters
+    ----------
+    store
+      A factory function providing a KeyValueStore
+    dataset_uuid
+      The dataset's unique identifier
+    table_name
+      Name of the kartothek table for which to retrieve the statistics
+    predicates
+      Kartothek predicates to apply filters on the data for which to gather statistics
+
+      .. warning::
+          Filtering will only be applied for predicates on indices.
+          The evaluation of the predicates therefore will therefore only return an approximate result.
+
+    frac
+      Fraction of the data to use for gathering statistics. `frac == 1` will use the whole dataset.
```

Comment:
That's slightly imprecise. It's the fraction of partitions, which has two implications:

- for very few partitions the effect of the fraction parameter is quite coarse
- for a skewed distribution (size over partitions) sampling large partitions in and out might effect the result more than for even distributions, so the user should be aware of that

Also: is the returned data what was collected or do you extrapolate based on the fraction (e.g. for `frac=0.1`, does the user get roughly 10% of the true dataset size or do you calculate `collected_size / frac` and return that)

Reply:
Regarding the normalization of the result you mentioned we also had this discussion and, quite frankly, I am not sure what the better API is. Theoretically everything is available for the user to normalize the data him/herself and with the implications / risks you already mentioned regarding too small sample sizes / inhomogeneous partitions, we figured that we shouldn't normalize it here.
otoh, normalizing it is currently a bit awkward since somewhere we need to get the total number of partitions which is available but not easily.

Input / suggestions welcome

Reply:
>  It's the fraction of partitions, which has two implications:
Good point, both implications. I think, I should extend the docstring a bit more.

I have thought of the interface as "Please give me metadata about `frac` % of my data" -- so no normalization. Perhaps naming this whole thing `collect_statistics` is misleading... We are not really doing any statistics (beyond the sampling and calculating average number of rows per row group) and essentially offer the user a possibility to easily collect and view metadata.

Reply:
Since you're likely returning a row per partition, I think it is the better API to NOT normalize here 👍 

Regarding the name: I think "statistics" is OK, at least I cannot think of anything better.

Reply:
AFAI see Parquet calls what we're accessing here "metadata", see https://parquet.apache.org/documentation/latest/